packages/mongodb-artifact-generator/src/commands/generateJiraPromptResponse.ts

```diff
@@ -0,0 +1,413 @@
+import {
+  createConfiguredAction,
+  withConfig,
+  withConfigOptions,
+} from "../withConfig";
+import { createCommand } from "../createCommand";
+import { makeRunLogger, type RunLogger } from "../runlogger";
+import path from "path";
+import { PromisePool } from "@supercharge/promise-pool";
+import { makeSummarizer, Summary } from "../chat/makeSummarizer";
+import { assertEnvVars } from "mongodb-rag-core";
+import { asBulletPoints, loadPromptExamplePairFromFile } from "../chat/utils";
+import { createRunId } from "../runId";
+import { makeGeneratePrompts } from "../chat/makeGeneratePrompts";
+import { makeGenerateResponse } from "../chat/makeGenerateResponse";
+import { GENERATOR_LLM_ENV_VARS } from "../ArtifactGeneratorEnvVars";
+import { promises as fs } from "fs";
+import { z } from "zod";
+
+const { GENERATOR_LLM_MAX_CONCURRENCY, OPENAI_CHAT_COMPLETION_DEPLOYMENT } =
+  assertEnvVars({
+    ...GENERATOR_LLM_ENV_VARS,
+    OPENAI_CHAT_COMPLETION_DEPLOYMENT: "",
+  });
+
+let logger: RunLogger;
+
+export interface JiraComment {
+  id: string;
+  body: string;
+  author: {
+    emailAddress: string;
+    displayName: string;
+  };
+  created: string;
+  updated: string;
+}
+
+export interface FormattedJiraIssue {
+  key: string;
+  projectName: string;
+  summary: string;
+  status: string;
+  created: string;
+  updated: string;
+  description: string;
+  comments: JiraComment[];
+}
+
+export type FormattedJiraIssueWithSummary = {
+  issue: FormattedJiraIssue;
+  summary: Summary;
+};
+
+type GenerateJiraPromptResponseCommandArgs = {
+  runId?: string;
+  llmMaxConcurrency: number;
+  maxInputLength?: number;
+  issuesFilePath?: string;
+  issue?: string | string[];
+};
+
+const issueKeysSchema = z.array(z.string());
+type IssueKeys = z.infer<typeof issueKeysSchema>;
+
+export default createCommand<GenerateJiraPromptResponseCommandArgs>({
+  command: "generateJiraPromptResponse",
+  builder(args) {
+    return withConfigOptions(args)
+      .option("runId", {
+        type: "string",
+        demandOption: false,
+        description:
+          "A unique name for the run. This controls where outputs artifacts and logs are stored.",
+      })
+      .option("llmMaxConcurrency", {
+        type: "number",
+        demandOption: false,
+        default: z.coerce
+          .number()
+          .default(10)
+          .parse(GENERATOR_LLM_MAX_CONCURRENCY),
+        description:
+          "The maximum number of concurrent requests to the LLM API. Defaults to 10. Can be specified in the config file as GENERATOR_LLM_MAX_CONCURRENCY.",
+      })
+      .option("maxInputLength", {
+        type: "number",
+        demandOption: false,
+        description:
+          "The maximum number of issues to process in this run. Any additional issues are skipped.",
+      })
```

Comment:
why is this functionality included?

Reply:
Purely for testing purposes - e.g. run my updated prompt on the first 6 of the top issues list.